modules/apps/callbacks/types/callbacks.go

```diff
@@ -0,0 +1,49 @@
+package types
+
+import (
+	channeltypes "github.com/cosmos/ibc-go/v7/modules/core/04-channel/types"
+	porttypes "github.com/cosmos/ibc-go/v7/modules/core/05-port/types"
+	ibcexported "github.com/cosmos/ibc-go/v7/modules/core/exported"
+)
+
+// PacketUnmarshalerIBCModule is an interface that combines the IBCModule and PacketDataUnmarshaler
+// interfaces to assert that the underlying application supports both.
+type PacketUnmarshalerIBCModule interface {
+	porttypes.IBCModule
+	porttypes.PacketDataUnmarshaler
+}
+
+// CallbackData is the callback data parsed from the packet.
+type CallbackData struct {
+	SrcContractAddr  string
+	DestContractAddr string
+	GasLimit         uint64
+	CustomMsg        []byte
```

Comment:
Should the custom msg the same for src and dest callbacks? Why does the src callback even need a custom msg in principle since it could store any info it needs before sending packet

Reply:
I've removed custom msg after feedback from confio. And I remember this wasn't in product requirements anyway